index.js

```diff
@@ -283,28 +286,84 @@ export const Config = {
 }
 
 export async function timer({ key }) {
+  const { state } = program;
   const [ owner, repo ] = key.split('/')
-  const result = await  client.activity.getEventsForRepo({ owner, repo });
-    for (let event of result.data) {
-      const { type, payload} = event;
-      if (type === 'IssuesEvent' && payload.action === 'opened') {
-        
-        // dispatch Event
-        const repoRef = root.users.one({ name: owner }).repos.one({ name: repo })
-        await repoRef.issueOpened.dispatch({
-          issue: repoRef.issues.one({ number: payload.issue.number })
-        });
-      };
-      if (type === 'PullRequestEvent' && payload.action === 'opened') {
-
-        // dispatch Event
-        const repoRef = root.users.one({ name: owner }).repos.one({ name: repo })
-        await repoRef.pullRequestOpened.dispatch({
-            issue: repoRef.issues.one({ number: payload.pull_request.number }),
-            pullRequest: repoRef.pullRequests.one({ number: payload.pull_request.number })
-        });
+  const result = await client.activity.getEventsForRepo({ owner, repo });
+
+  const data = result.data.reverse();
+  
+  const index = data
+    .findIndex(item => formatTime(item.created_at) > state.repos[key].lastEventTime);
+
+  const newEvents = data.splice(index);
+
+  for (let event of newEvents) {
+    const { type, payload } = event;
+    for (let event of state.repos[key].events) {
+      switch (event) {
+        case "issueOpened": {
+          if (type === "IssuesEvent" && payload.action === "opened") {
+            // dispatch Event
+            const repoRef = root.users.one({ name: owner }).repos.one({ name: repo });
+            await repoRef.issueOpened.dispatch({
+              issue: repoRef.issues.one({ number: payload.issue.number })
+            });
+          }
+        }
+        case "pullRequestOpened": {
+          if (type === "PullRequestEvent" && payload.action === "opened") {
+            // dispatch Event
+            const repoRef = root.users.one({ name: owner }).repos.one({ name: repo });
+            await repoRef.pullRequestOpened.dispatch({
+              issue: repoRef.issues.one({number: payload.pull_request.number}),
+              pullRequest: repoRef.pullRequests.one({number: payload.pull_request.number})
+            });
+          }
+        }
       }
     }
+  }
+  if (data.length > 0) {
```

Comment:
Shouldn't this check size of the filtered array instead? Maybe don't even define `const data` since it's not really needed